netket/utils/group/_group.py

```diff
@@ -204,7 +204,9 @@ def character_table_by_class(self) -> Array:
         # since we only want a random linear combination, we forget about the
         # constant |S| and only divide each column through with the appropriate |T|
         class_matrix = (
-            classes @ np.random.uniform(size=len(self))[self.product_table] @ classes.T
+            classes
+            @ random(len(self), jax.random.PRNGKey(0))[self.product_table]
```

Comment:
Yes, that's what is was thinking.

I suggest adding a comment here so people looking at this in the future understand what the purpose of the random vector is.

Reply:
I added some more comments outlining the whole algorithm at the start (there was already some here!) 
Could you draft something if you think more is needed?

Reply:
See my comment above.
To make it more 'transparent' I'm ok with fixing the PRNG seed but then let's just use numpy.*
It will also be faster.